src/ogdc_runner/recipe/simple.py

```diff
@@ -1,269 +1,170 @@
-            volume_mounts=[
-                models.VolumeMount(
-                    name=OGDC_WORKFLOW_PVC.name,
-                    mount_path="/mnt/",
-                ),
-            ],
-        )
-
-        with Steps(name="steps"):
-            check_dir_template()
-
-    # wait for the workflow to complete.
-    workflow_name = submit_workflow(workflow=w, wait=True)
-
-    # If overwrite is not True, we need to check the result of the
-    # `check-already-published` step to see if the data have been published or
-    # not.
-    # Check the result. Get an updated instance of the workflow, with the latest
-    # states for all notdes. Then, iterate through the nodes and find the
-    # template we define above ("check-already-published") and extract its
-    # output parameter.
-    completed_workflow = ARGO_WORKFLOW_SERVICE.get_workflow(name=workflow_name)
-    result = None
-    for node in completed_workflow.status.nodes.values():  # type: ignore[union-attr]
-        if node.template_name == "check-already-published":
-            result = node.outputs.parameters[0].value  # type: ignore[union-attr, index]
-    if not result:
-        err_msg = "Failed to check if data have been published"
-        raise OgdcWorkflowExecutionError(err_msg)
-
-    assert result in ("yes", "no")
-
-    # Cleanup this workflow, it is no longer needed
-    ARGO_WORKFLOW_SERVICE.delete_workflow(workflow_name)
-
-    return result == "yes"
-
-
-def data_already_published(*, recipe_config: RecipeConfig, overwrite: bool) -> bool:
-    """Check for the existence of published data for the given
-    recipe and optionally remove it.
-
-    If `overwrite=True`, this function will remove any existing published data
-    for the provided recipe.
-
-    Returns `True` if data have already been published for the given recipe,
-    otherwise `False`.
-    """
-    if overwrite:
-        # If `overwrite` is True, remove the existing data and return `False`.
-        _remove_existing_published_data(
-            recipe_config=recipe_config,
-        )
-        return False
-
-    return _check_for_existing_published_data(
-        recipe_config=recipe_config,
-    )
+from ogdc_runner.recipe.viz_workflow import submit_viz_workflow_recipe
 
 
 def make_and_submit_simple_workflow(
     recipe_config: RecipeConfig,
     wait: bool,
+    custom_image: str | None = None,
+    custom_tag: str | None = None,
+    custom_namespace: str | None = None,
+    update_global: bool = False,
 ) -> str:
     """Create and submit an argo workflow based on a simple recipe.
 
+    Args:
+        recipe_config: The recipe configuration
+        wait: Whether to wait for the workflow to complete
+        custom_image: Optional custom image to use for all containers
+        custom_tag: Optional custom tag for the image
+        custom_namespace: Optional custom namespace for the workflow
+        update_global: If True, update the global image config; if False, only apply to this workflow
```

Comment:
We might want to think more about if this pattern makes sense. I could see it quickly getting confusing about when and why/how we should apply a global update vs a workflow-specific update.

It may even be worth considering removing some config from being global at all (e.g., the docker image/tag) if it avoids this split/confusion.